### PR TITLE
Feature/212 apply unit test for add or update role

### PR DESCRIPTION
**Pull request by tuanph**

### CODE REVIEW

File: api/App.Test/Security/Role/UpdateRoleTest.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                Assert.IsTrue(false);
+            }
+            catch (ValidationException ex)
+            {
+                Assert.IsTrue(ex.HasExceptionKey("security.addOrUpdateRole.validation.nameAlreadyExisted"));
+            }
+        }
+
+        [TestMethod]
+        public void Security_Role_UpdateRole_ShouldGetException_WithDuplicatedKey()
```
Comment (tuanph) [inline api/App.Test/Security/Role/UpdateRoleTest.cs]: The key=name.replace(" ","_")

File: api/App.Service.Impl/Security/RoleService.cs [hunk 2]
```diff
@@ -167,20 +179,26 @@ private void ValidateUpdateRequest(UpdateRoleRequest request)
         {
             if (request.Id == null || request.Id == Guid.Empty)
             {
-                throw new ValidationException("security.addOrUpdateRole.validation.idIsInvalid");
+                throw new ValidationException("security.roles.validation.idIsInvalid");
```
Comment (tuanph) [inline api/App.Service.Impl/Security/RoleService.cs]: Yes, but I think we should change to "roleIdIsInvalid" to the same convention with Permission, Category

File: api/App.Service/Security/Role/UpdateRoleRequest.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
+﻿namespace App.Service.Security.Role
```
Comment (techcoaching) [inline api/App.Service/Security/Role/UpdateRoleRequest.cs]: Please remove unused name space

File: api/App.Service.Impl/Security/RoleService.cs [hunk 4]
```diff
@@ -98,17 +107,17 @@ private void ValidateDeleteRequest(Guid id)
         {
             if (id == null || id == Guid.Empty)
```
Comment (techcoaching) [inline api/App.Service.Impl/Security/RoleService.cs]: please use Required attribute for this

File: api/App.Service.Impl/Security/RoleService.cs [hunk 5]
```diff
@@ -165,28 +174,32 @@ private void RemoveRemovedPermissions(Role existedRole, UpdateRoleRequest reques
 
         private void ValidateUpdateRequest(UpdateRoleRequest request)
         {
+            IValidationException validationException = ValidationHelper.Validate(request);
             if (request.Id == null || request.Id == Guid.Empty)
```
Comment (techcoaching) [inline api/App.Service.Impl/Security/RoleService.cs]: please use Required attribute for this

**Pull request closed without merging**